src/CheckboxSelector/widget/checkboxselector.js

```diff
@@ -1,459 +1,418 @@
-			        tdDiv.innerHTML = currency;
-			       }
-			       break;
-			      case 'Yen' :
-			       if (displayAttr !== '') {
-			        currency = '&#165 '+mx.parser.formatNumber(displayAttr, splitobj.useSeparators, 2);
-			        tdDiv.innerHTML = currency;
-			       }
-			       break;
-			      case 'Pound' :
-			       if (displayAttr !== '') {
-			        currency = '&#163 '+mx.parser.formatNumber(displayAttr, splitobj.useSeparators, 2);
-			        tdDiv.innerHTML = currency;
-			       }
-			       break;
-			      default :
-			       break;
-			       // type not found
-			     }
-			    }, currObj, tdDiv));
+		publishObj: function (obj) {
+			if (this.listenchannel !== '')
+				connect.publish(this.getContent() + "/" + this.listenchannel + "/context", [obj]);
+		},
+		boxChanged: function (guid, checked) {
+			if (checked) {
+				this.context.addReference(this.assoc, guid);
+			} else {
+				this.context.removeReferences(this.assoc, [guid]);
 			}
 
-		}
-		/*
-		* Publish to formloader
-		*/
-		if (this.listenchannel !== '' && objs[0] && !this.beginEmpty)
-			dojo.publish(this.getContent() + "/"+this.listenchannel+"/context", [objs[0]]);
-		
-		surrDiv['tabIndex'] = 0;
-		this.domNode.appendChild(surrDiv);
-		this.validationDiv = mxui.dom.div({'display':'none'});
-		mxui.dom.addClass(this.validationDiv, 'mendixReferenceSetSelector_invalidNode');
-		this.domNode.appendChild(this.validationDiv);
-	},
-	publishObj : function (obj) {
-		if (this.listenchannel !== '')
-			dojo.publish(this.getContent() + "/"+this.listenchannel+"/context", [obj]);
-	},
-	boxChanged : function (guid, checked) {
-		if (checked) {
-			this.context.addReference(this.assoc, guid);
-		} else {
-			this.context.removeReferences(this.assoc, [guid]);
-		}
+			if (!this.ignoreChange) {
+				if (this.onchangemf) {
+					this.ignoreChange = true;
+					var context = new mendix.lib.MxContext();
+
+					if (this.context) {
+						context.setContext(this.context.getClass(), this.context.getGuid());
+					}
 
+					mx.ui.action(this.onchangemf, {
+						context: context,
+						callback: function () {
+						},
+						error: function (e) {
+							logger.error('Could not execute microflow: ' + this.onchangemf + 'with error: ' + e && e);
+						}
+					});
 
-		if(this.context.hasChanges()) {
-			this.context.save({
-				callback: function(){
-					//ok
-				},
-				error: function(e) {
-					logger.error('Could not save object: '+this.context+'with error: '+e&&e);
+					setTimeout(lang.hitch(this, function () {
+						this.ignoreChange = false;
+					}), 0);
 				}
-			});
-		}
+			}
+		},
 
-		if(!this.ignoreChange){
-			if(this.onchangemf) {
-				this.ignoreChange = true;
-				var context = mx.ui.newContext();
-				
-				if(this.context) {
-					context.setContext(this.context.getClass(), this.context.getGUID());
+		checkOrUncheckAll: function (e) {
+			var checkboxes = []
+			var objs = query('obj');
+			var allBoxes = query('input');
```

Comment:
Incorrect! your query will ook in the full document and will uncheck all checkboxes of your screen, including the ones that are not part of the widget :D

Provide a context to query. See https://dojotoolkit.org/api/
`query(selector,context);`

``` js
var objs = query('obj', this.domNode);
var allBoxes = query('input', this.domNode);
```